RAT/classlist.py

```diff
@@ -61,15 +61,22 @@ def __repr__(self):
                 output = repr(self.data)
         return output
 
-    def __setitem__(self, index: int, set_dict: dict[str, Any]) -> None:
-        """Assign the values of an existing object's attributes using a dictionary."""
-        self._setitem(index, set_dict)
+    def __setitem__(self, index: int, item: Union['RAT.models', dict[str, Any]]) -> None:
+        """Assign the values of an existing object's attributes using either a replacement object or a dictionary
+        containing key-value pairs.
+        """
+        self._setitem(index, item)
 
-    def _setitem(self, index: int, set_dict: dict[str, Any]) -> None:
+    def _setitem(self, index: int, item: Union['RAT.models', dict[str, Any]]) -> None:
```

Comment:
I think it might be better to support one type i.e only models or dict (my pick is models) then provide a fallback for anyone who wants to use the alternate e.g. add_rows_as_type. Otherwise there will be failing edge cases in the current design

Reply:
Yeah, that's fine with me. I'll get on it!